#168781621 - Signup Users

nignanthomas opened the pull request:
#### What does this PR do?
Signup Users

#### Description of Task to be completed?
- Create global components Button, ToastContainer
- Create signupActions and Reducers
- Create Register view

#### How should this be manually tested?
- Clone ft-signup-168781621 branch
- run `npm i`
- run `npm run dev`
- navigate to localhost:<:port>/users
- Fill the login form
- `Register`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168781621](https://www.pivotaltracker.com/story/show/168781621)

#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2019-12-07 at 18 04 00" src="https://user-images.githubusercontent.com/42959540/70377364-64d9de80-191c-11ea-8db7-3718662acbde.png">
<img width="439" alt="Screen Shot 2019-12-07 at 18 04 47" src="https://user-images.githubusercontent.com/42959540/70377366-67d4cf00-191c-11ea-92de-9265d557da59.png">

#### Questions:
N/A

nignanthomas committed:
feature(signup): Signup Users

- create global components Button, ToastContainer

- create signupActions and Reducers

- create Register view

[Finishes #168781621]

alainmateso reviewed (changes requested):
Great job Thomas, Kindly return a descriptive message when a user enters data that don't match the required pattern.

ngireric123 commented:
Great job @nignanthomas kindly try to increase test coverage

ngireric123 reviewed (changes requested):
@nignanthomas kindly  increase test coverage

nakiwuge commented on src/actions/SignupAction.js:
```diff
@@ -0,0 +1,54 @@
+import axios from 'axios';
+import { SIGNUP_PENDING, SIGNUP_SUCCESS, SIGNUP_FAIL } from './types';
+
+const appUrl = 'https://caret-bn-backend-staging.herokuapp.com/api/v1';
```
Configure axios with  the baseUrl

nakiwuge commented on src/actions/SignupAction.js:
```diff
@@ -0,0 +1,54 @@
+  });
+};
+
+// export default { signupAction, resetSignup };
```
Make this file to have all auth Actions. Create one type function to be used in all the auth action functions. something like this
```
const authType = (type,payload)=>{
type,
payload
}
```
Make signUp action cleaner with something like this
```
  axios.post(`${appUrl}/users/register`, userData)
    .then((response) => {
        dispatch(authType(SIGNUP_SUCCESS,response))
    })
    .catch((error) => {
 dispatch(authType(SIGNUP_FAIL,error.reponse))
    });
};
```

nakiwuge commented on src/actions/SignupAction.js:
```diff
@@ -0,0 +1,54 @@
+  });
+  axios.post(`${appUrl}/users/register`, userData)
+    .then((response) => {
+      dispatch({
```
Kindly create an `isLoading` state in the component where you call this action to cater for loading

nakiwuge commented on src/actions/SignupAction.js:
```diff
@@ -0,0 +1,54 @@
+};
+
+// export default { signupAction, resetSignup };
+// module.exports = { signupAction, resetSignup };
```
Remove commented code

nakiwuge commented on src/components/features/Signup.js:
```diff
@@ -0,0 +1,89 @@
+      email, username, password, confirmPassword,
+    } = this.state;
+
+    const handleChange = (e) => {
```
Move this method out of the render

nakiwuge commented on src/reducers/signupReducer.js:
```diff
@@ -0,0 +1,33 @@
+const initialState = {
+  signupPending: false,
+  signupFail: null,
+  signupSuccess: false,
```
rename `signupSuccess` to `data: null`

nakiwuge commented on src/reducers/signupReducer.js:
```diff
@@ -0,0 +1,33 @@
+  }
+};
+
+// export default signupReducer;
```
remove commented code

nakiwuge commented on src/styles/styles.scss:
```diff
@@ -1,21 +1,102 @@
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+    > div{
```
remove `>`

nakiwuge commented on src/styles/styles.scss:
```diff
@@ -1,21 +1,102 @@
+			margin: 0.2rem 0.5rem 1rem 0;
+			transition: all 0.5s ease-in-out;
+			height: 2rem;
+			width: 150px;
```
Indent button

nakiwuge commented on src/styles/styles.scss:
```diff
@@ -1,21 +1,102 @@
+    }
+}
+.login-form, .signup-form{
+    > form Input{
```
use sass here like this 
``` form{
Input{
        width: 80%;
        padding-left: 0.5rem;
    }
} 
```

nakiwuge commented on src/tests/SignupTest.spec.js:
```diff
@@ -0,0 +1,38 @@
+import { Signup } from '../components/features/Signup';
+
+const mockFunc = jest.fn();
+
```
remove this line

nakiwuge commented on src/tests/SignupTest.spec.js:
```diff
@@ -0,0 +1,38 @@
+  });
+
+  // it('Should render ', () => {
+  //   expect(wrapper.find('').text()).toBe('');
```
remove this commented code

nakiwuge commented on src/tests/actionsTests/SignupActionsTest.spec.js:
```diff
@@ -0,0 +1,145 @@
+import { SIGNUP_PENDING, SIGNUP_SUCCESS, SIGNUP_FAIL } from '../../actions/types';
+import { signupAction, resetSignup } from '../../actions/SignupAction';
+// import store from '../../reduxStore';
```
remove this commented code

nakiwuge commented on src/tests/actionsTests/SignupActionsTest.spec.js:
```diff
@@ -0,0 +1,145 @@
+// import store from '../../reduxStore';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+
```
remove this line

nignanthomas committed:
feature(signup): Signup Users

- add OR separator between form and social auth

- make the page mobile responsive

[Delivers #168781621]

ngireric123 reviewed (approved):
LGTM
